lib/token-util.js

```diff
@@ -0,0 +1,36 @@
+var assign = require('lodash.assign')
+
+// set/replace or append to an attribute on a markdown-it Token
+function tokenAttr (token, attr, value, operation) {
+  var idx = token.attrIndex(attr)
+  if (idx < 0) {
+    token.attrPush([attr, value])
+  } else {
+    if (operation === 'append') {
+      token.attrs[idx][1] += ' ' + value
+    } else {
+      token.attrs[idx][1] = value
+    }
+  }
+}
+
+// Look for tokens with type === 'emoji', which are :shortcode: style emoji
+// characters that have been replaced by the markdown-it-emoji plugin. Return the
+// original tokens, unless they were converted gemoji strings; then return a copy so
+// we haven't clobbered the original when it comes time to render HTML
+function unconvertEmoji (Token) {
+  return function (token) {
+    var tok = token
+    if (token.type === 'emoji') {
+      tok = assign(new Token(), token)
+      tok.meta = token.content
```

Comment:
I just realized I don't need this line anymore now that I'm using lodash.assign to make copies; was an artifact of an earlier attempt. I'll remove it when I get a chance.